src/utils/ability-utils.ts

```diff
@@ -44,3 +44,23 @@ export function getWeatherCondition(...weatherTypes: WeatherType[]): AbAttrCondi
 export const normalTypeMoveConversionCondition: PokemonAttackCondition = (user, _target, move) =>
   move?.type === ElementalType.NORMAL
   && (!move.hasAttr(VariableMoveTypeAttr) || (TERA_MOVES.includes(move.id) && !user?.isTerastallized));
+
+/** Abilities perceived by the Enemy AI to have high value */
+export const highValueAbilities: Readonly<AbilityId[]> = [
+  AbilityId.WONDER_GUARD,
+  AbilityId.DESOLATE_LAND,
+  AbilityId.PRIMORDIAL_SEA,
+  AbilityId.HUGE_POWER,
+  AbilityId.PURE_POWER,
+  AbilityId.CONTRARY,
+];
+
+/** Abilities perceived by the Enemy AI to have a detrimental effect on the source */
+export const detrimentalAbilities: Readonly<AbilityId[]> = [
```

Comment:
NIT: We agree on `UPPER_SNAKE_CASE` for constants.
They should also be put into a `src/constants/ability-constants.ts` file.
I'd also use `Object.freeze([..])` to actually freeze the object, as types are _dropped_ upon bundling to js